internal/types/subtypes/registry_global.go

```diff
@@ -111,6 +117,62 @@ func (r *registry) register(ctx context.Context, domain string, subtype Subtype,
 	return nil
 }
 
+func (r *registry) registerRequestTransformationFunc(ctx context.Context, msg proto.Message, transformFn TransformationFunc) error {
+	r.Lock()
+	defer r.Unlock()
+
+	fqn := msg.ProtoReflect().Descriptor().FullName()
+	const op = "subtypes.(Registry).RegisterRequestTransformationFunc"
+	if _, present := r.requestTransformationFuncs[fqn]; present {
+		return errors.New(
+			ctx,
+			errors.SubtypeAlreadyRegistered,
+			op,
+			fmt.Sprintf("proto type %q already has a request transformation function registered", fqn),
+		)
+	}
+	r.requestTransformationFuncs[fqn] = transformFn
+	return nil
+}
+
+func (r *registry) registerResponseTransformationFunc(ctx context.Context, msg proto.Message, transformFn TransformationFunc) error {
+	r.Lock()
+	defer r.Unlock()
+
+	fqn := msg.ProtoReflect().Descriptor().FullName()
+	const op = "subtypes.(Registry).RegisterResponseTransformationFunc"
+	if _, present := r.responseTransformationFuncs[fqn]; present {
+		return errors.New(
+			ctx,
+			errors.SubtypeAlreadyRegistered,
+			op,
+			fmt.Sprintf("proto type %q already has a response transformation function registered", fqn),
+		)
+	}
+	r.responseTransformationFuncs[fqn] = transformFn
+	return nil
+}
+
+func transformRequest(msg proto.Message) error {
+	fqn := msg.ProtoReflect().Descriptor().FullName()
+	if fn, ok := globalRegistry.requestTransformationFuncs[fqn]; ok {
+		return fn(msg)
+	}
+	return transformRequestAttributes(msg)
+}
+
+func transformResponse(msg proto.Message) error {
+	fqn := msg.ProtoReflect().Descriptor().FullName()
+	if fn, ok := globalRegistry.responseTransformationFuncs[fqn]; ok {
+		return fn(msg)
+	}
+	return transformResponseAttributes(msg)
+}
```

Comment:
i think these functions should be in `interceptor.go`, as it is now, trying to trace the logic from the interceptor brings you to this file, which then goes back to `interceptory.go` for `transformResponseAttributes`

Reply:
Makes sense 👍🏻

Reply:
Fixed